widget/list.go

```diff
@@ -60,22 +76,53 @@ func (l *List) MinSize() fyne.Size {
 	return l.BaseWidget.MinSize()
 }
 
+// Selection returns the index of the data that is currently selected or -1 if there is no selection.
+func (l *List) Selection() int {
+	return l.selected
+}
+
+// SetSelection sets the index of the data that is to be selected, refreshes, and calls the OnSelected callback.
+func (l *List) SetSelection(index int) {
+	length := 0
+	if f := l.Length; f != nil {
+		length = f()
+	}
+	if index < 0 || index >= length {
+		return
+	}
+	l.selected = index
+	if index >= l.firstItemIndex && index <= l.lastItemIndex {
+		l.Refresh()
+		return
```

Comment:
This return means OnSelectionChanged will not be triggered if the selection is in viewport

Reply:
Oopsy great eye. Fixed.